src/pyOpenMS/pyopenms/addons/ximparquetfile.py

```
@@ -0,0 +1,337 @@
+                "pandas is required for to_df(). Install with `pip install pandas`."
+            ) from e
+        data = self.to_dict(explode=explode)
+        return pd.DataFrame({k: list(v) for k, v in data.items()})
+
+
+@addon("XIMParquetFile")
+def __len__(self):
+    """Return the number of files associated with this instance."""
+    try:
+        return len(self.getFilenames())
+    except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)